src/System.Configuration/src/System/Configuration/RuntimeConfigurationRecord.cs

```diff
@@ -0,0 +1,192 @@
+                            section.SectionInformation.SectionName), e);
+                }
+            }
+
+            return runtimeObject;
+        }
+
+        private class RuntimeConfigurationFactory
+        {
+            private ConstructorInfo _sectionCtor;
+            private IConfigurationSectionHandler _sectionHandler;
+
+            internal RuntimeConfigurationFactory(RuntimeConfigurationRecord configRecord, FactoryRecord factoryRecord)
+            {
+                Init(configRecord, factoryRecord);
+            }
+
+            private void Init(RuntimeConfigurationRecord configRecord, FactoryRecord factoryRecord)
+            {
+                // Get the type of the factory
+                Type type = TypeUtil.GetType(configRecord.Host, factoryRecord.FactoryTypeName,
+                    true);
+
+                // If the type is a ConfigurationSection, that's the type.
+                if (typeof(ConfigurationSection).IsAssignableFrom(type))
+                {
+                    _sectionCtor = TypeUtil.GetConstructor(type, typeof(ConfigurationSection),
+                        true);
+                }
+                else
+                {
+                    // Note: in v1, IConfigurationSectionHandler is in effect a factory that has a Create method
+                    // that creates the real section object.
+
+                    // throws if type does not implement IConfigurationSectionHandler
+                    TypeUtil.VerifyAssignableType(typeof(IConfigurationSectionHandler), type, true);
+
+                    // Create an instance of the handler
+                    _sectionHandler =
+                        (IConfigurationSectionHandler)TypeUtil.CreateInstance(type);
+                }
+            }
+
+            // Throw an exception if an attribute within a legacy section is one of our
+            // reserved locking attributes. We do not want admins to think they can lock
+            // an attribute or element within a legacy section.
+            private static void CheckForLockAttributes(string sectionName, XmlNode xmlNode)
+            {
+                XmlAttributeCollection attributes = xmlNode.Attributes;
+                if (attributes != null)
+                {
+                    foreach (XmlAttribute attribute in attributes)
+                        if (ConfigurationElement.IsLockAttributeName(attribute.Name))
+                        {
+                            throw new ConfigurationErrorsException(
+                                string.Format(SR.Config_element_locking_not_supported, sectionName), attribute);
+                        }
+                }
+
+                foreach (XmlNode child in xmlNode.ChildNodes)
+                    if (xmlNode.NodeType == XmlNodeType.Element) CheckForLockAttributes(sectionName, child);
+            }
+
+            internal object CreateSection(bool inputIsTrusted, RuntimeConfigurationRecord configRecord,
```

Comment:
I'll leave that for a separate item. I did it once then backed out of it- was a little concerned about potential roundtripping issues. I'll create an issue and follow up once I can write some validating tests.